# Travel, Tourism & Hospitality

## Customer Retention and Dynamic Pricing Analysis

### Week 1 - Day 7

GitHub Documentation and Week 1 Summary

Dataset:
Hotel Booking Demand Dataset

Author:
Niveditha J

In [1]:
import pandas as pd

df = pd.read_csv(
    r"C:\Users\Hp\Desktop\Travel-Tourism-Customer-Retention-Analysis\data\processed\hotel_bookings_validated.csv"
)

df.shape

(87359, 36)

## Week 1 Accomplishments

The first week focused on building a clean and reliable dataset for future analysis.

Major activities included:

- Data loading and understanding
- Missing value treatment
- Feature engineering
- ADR outlier detection
- ADR outlier treatment
- Dataset validation

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87359 entries, 0 to 87358
Data columns (total 36 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   hotel                           87359 non-null  object 
 1   is_canceled                     87359 non-null  int64  
 2   lead_time                       87359 non-null  int64  
 3   arrival_date_year               87359 non-null  int64  
 4   arrival_date_month              87359 non-null  object 
 5   arrival_date_week_number        87359 non-null  int64  
 6   arrival_date_day_of_month       87359 non-null  int64  
 7   stays_in_weekend_nights         87359 non-null  int64  
 8   stays_in_week_nights            87359 non-null  int64  
 9   adults                          87359 non-null  int64  
 10  children                        87359 non-null  float64
 11  babies                          87359 non-null  int64  
 12  meal                            

In [3]:
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

Rows: 87359
Columns: 36


## Engineered Features Created

Five additional features were developed to improve analytical capability.

1. total_stay_nights
2. total_guests
3. booking_changed
4. has_previous_cancellation
5. special_request_flag

In [4]:
engineered_features = [
    'total_stay_nights',
    'total_guests',
    'booking_changed',
    'has_previous_cancellation',
    'special_request_flag'
]

df[engineered_features].head()

,total_stay_nights,total_guests,booking_changed,has_previous_cancellation,special_request_flag
0,0,2.0,1,0,0
1,0,2.0,1,0,0
2,1,1.0,0,0,0
3,1,1.0,0,0,0
4,2,2.0,0,0,1


## ADR Outlier Treatment Summary

ADR was analyzed using the IQR method.

Upper Bound = 227

Outlier records were capped instead of removed.

This approach preserved all booking records while reducing the influence of extreme ADR values.

In [6]:
df['adr'].describe()

count    87359.000000
mean       105.340171
std         49.167077
min         -6.380000
25%         72.000000
50%         98.100000
75%        134.000000
max        227.000000
Name: adr, dtype: float64

## Data Quality Summary

Data quality checks were performed before starting Exploratory Data Analysis.

Validation Results:

- Missing Values = 0
- Duplicate Records = 0
- ADR Outliers Treated
- Engineered Features Verified

In [7]:
print("Missing Values:", df.isnull().sum().sum())
print("Duplicates:", df.duplicated().sum())

Missing Values: 0
Duplicates: 0


In [8]:
import os

os.makedirs(
    r"C:\Users\Hp\Desktop\Travel-Tourism-Customer-Retention-Analysis\data\processed",
    exist_ok=True
)

df.to_csv(
    r"C:\Users\Hp\Desktop\Travel-Tourism-Customer-Retention-Analysis\data\processed\week1_completed_dataset.csv",
    index=False
)

## Week 1 Findings

- Dataset successfully cleaned and validated.
- Missing values were eliminated.
- 11 duplicate records were removed.
- Five engineered features were created.
- ADR outliers were treated using capping.
- Final dataset contains 87,359 records and 36 columns.
- Dataset is ready for Week 2 Exploratory Data Analysis.

## Week 1 Conclusion

The data preparation phase was successfully completed.

A clean and validated hospitality booking dataset was prepared for advanced exploratory analysis.

The project is now ready to enter Week 2, where detailed Exploratory Data Analysis (EDA) will be performed to identify cancellation patterns, customer segments, booking trends, and dynamic pricing opportunities.